// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.customer";
option java_outer_classname = "CustomerProto";
option objc_class_prefix = "HLW";

package customer;

// The greeting service definition.
service Customer {
  // Sends a greeting
  rpc CreateUser (CreateUserRequest) returns (Confirmation) {}
  rpc ChangeLogin (ChangeLoginRequest) returns (Confirmation) {}
  rpc CheckLogin (CheckLoginRequest) returns (CheckLoginResponse) {}
  rpc GetRating (GetRatingRequest) returns (GetRatingResponse) {}
  rpc UpdateCart (UpdateCartRequest) returns (Confirmation) {}
  rpc GetShoppingCart (CheckLoginRequest) returns (GetShoppingCartResponse) {}
  rpc GetHistory (CheckLoginRequest) returns (GetShoppingCartResponse) {}
  rpc MakePurchase (CheckLoginRequest) returns (Confirmation) {}
}

message GetShoppingCartResponse {
  bool status = 1;
  repeated int32 item_ids = 2;
  repeated int32 quantities = 3;
  string error = 4;
}

message UpdateCartRequest {
  string username = 1;
  bool add = 2;
  string key = 3;
  int32 item_id = 4;
  int32 quantity = 5;
}

message GetRatingRequest {
  int32 seller_id = 1;
}

message GetRatingResponse {
  bool status = 1;
  int32 thumbsup = 2;
  int32 thumbsdown = 3;
  string error = 4;
}

// The request message containing the user's name.
message CreateUserRequest {
  string name = 1;
  string username = 2;
  string password = 3;
}

message ChangeLoginRequest {
  string username = 1;
  string password = 2;
  bool logging_in = 3;
}

message CheckLoginRequest {
  string username = 1;
}

message CheckLoginResponse {
  bool status = 1;
  bool logged_in = 2;
  string error = 3;
}

// The response message containing the greetings
message Confirmation {
  bool status = 1;
  string error = 2;
}
